### PR TITLE
[KIWI - 933] - Double clicking on Confirmation Button in CIC results in Error Page

### PR DESCRIPTION
**Pull request by Stelios-CB**
### What changed
Added new logic to /claimedId lambda to handle multiple requests made using the same session ID.

### Why did it change
Two bugs were occurring on the 'Confirm your details' page:

1. User clicks confirm button multiple times
2. User confirms details, then navigates back to 'Confirm your details page' and tries to confirm again

Both situations result in /claimedId lambda returning wrong session state error

### Issue tracking
[KIWI - 933](https://govukverify.atlassian.net/browse/KIWI-933)

### Evidence:
Error 1:
CIC error 1

Error 2:
CIC error 2

Fix:
Fix

### CODE REVIEW

File: src/services/ClaimedIdRequestProcessor.ts [hunk 1]
```diff
@@ -75,19 +76,25 @@ export class ClaimedIdRequestProcessor {
...
-				});
-				return new Response(HttpCodesEnum.UNAUTHORIZED, `Session is in the wrong state: ${session.authSessionState}`);
+			switch (session.authSessionState) {
+			  case AuthSessionState.CIC_SESSION_CREATED:
+					await this.cicService.saveCICData(sessionId, cicSession, session.expiryDate);
+					return new Response(HttpCodesEnum.OK, "");
+			  case AuthSessionState.CIC_DATA_RECEIVED:
+			  case AuthSessionState.CIC_AUTH_CODE_ISSUED:
+			  case AuthSessionState.CIC_ACCESS_TOKEN_ISSUED:
+					this.logger.info(`Duplicate request for session with id: ${sessionId}, returning status 200`);
```
Comment (ViveakV) [inline src/services/ClaimedIdRequestProcessor.ts:86]: Here can we not include sessionID in the string but as a speedster key value please, makes it easier to search this log in splunk 

So keep log messages static strings

Comment (ViveakV) [reply src/services/ClaimedIdRequestProcessor.ts:86]: this.logger.info("Duplicate request for session, returning 200", sessionId);

File: src/services/ClaimedIdRequestProcessor.ts [hunk 2]
```diff
@@ -75,19 +76,22 @@ export class ClaimedIdRequestProcessor {
...
-				return new Response(HttpCodesEnum.UNAUTHORIZED, `Session is in the wrong state: ${session.authSessionState}`);
+			switch (session.authSessionState) {
+			  case AuthSessionState.CIC_SESSION_CREATED:
+					await this.cicService.saveCICData(sessionId, cicSession, session.expiryDate);
+					return new Response(HttpCodesEnum.OK, "");
+			  case AuthSessionState.CIC_DATA_RECEIVED:
+			  case AuthSessionState.CIC_AUTH_CODE_ISSUED:
+			  case AuthSessionState.CIC_ACCESS_TOKEN_ISSUED:
+					this.logger.info("Duplicate request, returning status 200, sessionId: ", sessionId);
+					return new Response(HttpCodesEnum.OK, "");
```
Comment (Caitlin-cooling) [inline src/services/ClaimedIdRequestProcessor.ts:87]: Could we add a message to these responses - maybe something like "Request already processed" and "Request successfully processed"?

Comment (Stelios-CB) [reply src/services/ClaimedIdRequestProcessor.ts]: Done

**Review by ViveakV — APPROVED** [review]

**Pull request merged**